### ISSUE TITLE
Default PML configuration disables $\kappa$ spatial variance

### ISSUE THREAD

**Issue by smartalecH** [NanoComp/meep#2277]
Going through some of the PML code, I noticed that the default `mean_stretch=1` effectively disables the $\kappa$ position-dependent behavior in the complex scaling coefficient,

$$
s = \kappa + \frac{i\sigma}{\alpha + \omega}
$$

https://github.com/NanoComp/meep/blob/b00f2d4ce20e683275a4895c8dc8318677a296d1/src/structure.cpp#L633

https://github.com/NanoComp/meep/blob/b00f2d4ce20e683275a4895c8dc8318677a296d1/src/structure.cpp#L675

This effectively sets $\kappa=1$. I'm assuming this is intentional, as all the [meep PML notes](https://github.com/NanoComp/meep/blob/b00f2d4ce20e683275a4895c8dc8318677a296d1/src/structure.cpp#L675) were written assuming this to be the case. Can we elaborate on the historical context behind this default? I'm assuming it has a good reason, especially since all of the `step_curl` stencils support an arbitrary $\kappa$ (a massive undertaking).

**Comment by stevengj** [comment]
Yes.   In general, increasing $\kappa$ makes the reflection worse AFAIK.   It's only useful for relatively specialized cases involving evanescent waves (which aren't normally a concern since they decay exponentially anyway).
